akka-http/src/main/scala/akka/http/scaladsl/server/directives/CodingDirectives.scala

```diff
@@ -107,6 +108,16 @@ trait CodingDirectives {
    */
   def decodeRequest: Directive0 =
     decodeRequestWith(DefaultCoders: _*)
+
+  /**
+   * Inspects the response entity and adds a `Content-Encoding: gzip` response header if
+   * the entities media-type is precompressed with gzip and no `Content-Encoding` header is present yet.
```

Comment:
Yes, it's stackable. But the order matters. Since someone has already added a `Content-Encoding` header, do we now add `gzip` to the front or the back?
I'd say, if someone added a `Content-Encoding` we simply assume she knows what she is doing and bail out...